sdk/push/src/main/java/com/klaviyo/push/KlaviyoPushService.kt

```diff
@@ -34,70 +32,68 @@ class KlaviyoPushService : FirebaseMessagingService() {
          * @param pushToken The push token provided by the FCM Service
          */
         fun setPushToken(pushToken: String) {
-            val properties = KlaviyoCustomerProperties()
-                .addAppendProperty(REQUEST_PUSH_KEY, pushToken)
+            val profile = Profile().addAppendProperty(PUSH_TOKEN_APPEND_KEY, pushToken)
 
-            Klaviyo.identify(properties)
+            Klaviyo.setProfile(profile)
 
-            KlaviyoPreferenceUtils.writeStringPreference(PUSH_TOKEN_PREFERENCE_KEY, pushToken)
+            Klaviyo.Registry.dataStore.store(PUSH_TOKEN_KEY, pushToken)
         }
 
         /**
-         * Returns the current push token that we have stored on this device
+         * Retrieve the device FCM push token  have stored on this device
          *
          * @return The push token we read from the shared preferences
          */
-        fun getCurrentPushToken(): String {
-            return KlaviyoPreferenceUtils.readStringPreference(PUSH_TOKEN_PREFERENCE_KEY) ?: ""
+        internal fun getPushToken(): String {
+            return Klaviyo.Registry.dataStore.fetch(PUSH_TOKEN_KEY) ?: ""
         }
 
         /**
          * Logs an $opened_push event for a remote notification that originated from Klaviyo
          *
-         * TODO customer and event properties should both be optional, once SDK is tracking customer state internally
+         * @param payload The data attributes of the push notification payload
+         */
+        internal fun openedPush(payload: Map<String, String>) {
+            payload["_k"] ?: return // Track only pushes originating from klaviyo
+
+            val event = Event().also {
+                payload.forEach { (k, v) -> it.setProperty(k, v) }
+                it.setProperty(PUSH_TOKEN_KEY, getPushToken())
+            }
+
+            Klaviyo.createEvent(KlaviyoEventType.OPENED_PUSH, event)
+        }
+
+        /**
+         * Handles a push received while app is in the foreground
          *
-         * @param pushPayload The data attributes of the push notification payload
-         * @param customerProperties Profile with which to associate the event
-         * @param eventProperties Optional additional properties for the event
+         * @param message
```

Comment:
Description for this param? Seems like you did the others so just noticed this one was missing